### PR TITLE
Added autofillHints attribute for text fields

### PR DESCRIPTION
**Pull request by hemish11**
Closes https://github.com/EnsembleUI/ensemble/issues/1495

Added the autofillHints attribute for the TextInput and also added autofillEnabled for the OTP field

### CODE REVIEW

File: modules/ensemble/lib/widget/input/form_textfield.dart [hunk 1]
```diff
@@ -156,6 +156,8 @@ abstract class BaseTextInput extends StatefulWidget
       'maxLines': (value) =>
           _controller.maxLines = Utils.optionalInt(value, min: 1),
       'textStyle': (style) => _controller.textStyle = Utils.getTextStyle(style),
+      'autofillHints': (value) =>
```
Comment (kmahmood74) [inline modules/ensemble/lib/widget/input/form_textfield.dart:159]: @hemish11 this won't work when someone is setting `autofillHints` in js. 

```js
myField.autofillHints = ['a','b'];
```
look at the implementation of `Utils.getListOfStrings` it's terrible code that returns null if it is not yaml. 

Please ALWAYS test the js property setting too. Your code is hardcoded to work only for yaml

**Review by kmahmood74 — CHANGES_REQUESTED** [review]

File: modules/ensemble/lib/util/utils.dart [hunk 2]
```diff
@@ -287,7 +287,7 @@ class Utils {
   }
 
   static List<String>? getListOfStrings(dynamic value) {
-    if (value is YamlList) {
+    if (value is YamlList || value is List) {
```
Comment (kmahmood74) [inline modules/ensemble/lib/util/utils.dart:290]: @hemish11 YamlList is List, isn't it? so the multiple check is not needed.

**Review by kmahmood74 — APPROVED** [review]
please look at my comment about not needing the YamlList check. Change if needed. Approved

File: modules/ensemble/lib/util/utils.dart (hunk 2 above)
Comment (hemish11) [reply modules/ensemble/lib/util/utils.dart:290]: Umm I did take a look at the implementation of YamlList and we do need this check, so ya closing it without changing it

**Pull request merged**